[WebGPUSwift] release build workaround for rdar://140993460

nmahendru opened the pull request:
#### f1903ebc9003743e7dde2bfc07f24c50f8815560
<pre>
[WebGPUSwift] release build workaround for <a href="https://rdar.apple.com/140993460">rdar://140993460</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=284462">https://bugs.webkit.org/show_bug.cgi?id=284462</a>
<a href="https://rdar.apple.com/141282776">rdar://141282776</a>

Reviewed by Geoffrey Garen.

Reordering inheritance list for CommandEncoder helps the release build to work until <a href="https://rdar.apple.com/140993460">rdar://140993460</a> is resolved.

* Source/WebGPU/WebGPU/CommandEncoder.h:

Canonical link: <a href="https://commits.webkit.org/287889@main">https://commits.webkit.org/287889@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/184df156140eda3a4450ae68dd4f21c40d73e5d5

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/80491 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/59498 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/34162 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/85012 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/31473 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/68559 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/7802 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/62897 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/20703 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/83560 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/52990 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/73283 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/43200 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/50293 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/29932 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/71437 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/27962 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/86446 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/7717 "Built successfully") | [⏳ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/macOS-Sequoia-Debug-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/71186 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/7892 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/69120 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/70425 "Found 5 new API test failures: /WebKitGTK/TestWebKitWebView:/webkit/WebKitWebView/submit-form, /WebKitGTK/TestWebProcessExtensions:/webkit/WebKitWebProcessExtension/dom-input-element-is-user-edited, /TestWebKit:WebKit.EnumerateDevicesCrash, /TestWebKit:WebKit2TextFieldBeginAndEditEditingTest.TextFieldDidBeginAndEndEditingEvents, /TestWebKit:WebKit.FocusedFrameAfterCrash (failure)") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/14428 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/13375 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/12589 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/7679 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/7518 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/11037 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/9323 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

geoffreygaren reviewed (approved):
r=me

Let's leave a comment behind with a Radar link, so mechanical refactoring doesn't revert this change by accident.

nmahendru committed:
[WebGPUSwift] release build workaround for rdar://140993460
https://bugs.webkit.org/show_bug.cgi?id=284462
rdar://141282776

Reviewed by Geoffrey Garen.

Reordering inheritance list for CommandEncoder helps the release build to work until rdar://140993460 is resolved.

* Source/WebGPU/WebGPU/CommandEncoder.h:

Canonical link: https://commits.webkit.org/287889@main
